core/src/tx/mod.rs

```diff
@@ -89,27 +91,70 @@ pub fn validate<Call: Payload>(call: &Call, metadata: &Metadata) -> Result<(), E
     Ok(())
 }
 
+/// Returns the suggested transaction versions to build for a given chain, or an error
+/// if Subxt doesn't support any version expected by the chain.
+///
+/// If the result is [`TransactionVersion::V4`], use the `v4` methods in this module. If it's
+/// [`TransactionVersion::V5`], use the `v5` ones.
+pub fn suggested_version(metadata: &Metadata) -> Result<TransactionVersion, Error> {
+    let versions = metadata.extrinsic().supported_versions();
+
+    if versions.contains(&4) {
+        Ok(TransactionVersion::V4)
+    } else if versions.contains(&5) {
+        Ok(TransactionVersion::V5)
+    } else {
+        Err(ExtrinsicError::UnsupportedVersion.into())
+    }
+}
+
+/// The transaction versions supported by Subxt.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum TransactionVersion {
+    /// v4 transactions (signed and unsigned transactions)
+    V4,
+    /// v5 transactions (bare and general transactions)
+    V5,
+}
+
 /// Return the SCALE encoded bytes representing the call data of the transaction.
 pub fn call_data<Call: Payload>(call: &Call, metadata: &Metadata) -> Result<Vec<u8>, Error> {
     let mut bytes = Vec::new();
     call.encode_call_data_to(metadata, &mut bytes)?;
     Ok(bytes)
 }
 
-/// Creates an unsigned extrinsic without submitting it.
-pub fn create_unsigned<T: Config, Call: Payload>(
+/// Creates a V4 "unsigned" transaction without submitting it.
+pub fn create_v4_unsigned<T: Config, Call: Payload>(
+    call: &Call,
+    metadata: &Metadata,
+) -> Result<Transaction<T>, Error> {
+    create_unsigned_at_version(call, 4, metadata)
+}
+
+/// Creates a V5 "bare" transaction without submitting it.
+pub fn create_v5_bare<T: Config, Call: Payload>(
     call: &Call,
     metadata: &Metadata,
+) -> Result<Transaction<T>, Error> {
+    create_unsigned_at_version(call, 5, metadata)
+}
+
+// Create a V4 "unsigned" transaction or V5 "bare" transaction.
+fn create_unsigned_at_version<T: Config, Call: Payload>(
+    call: &Call,
+    tx_version: u8,
+    metadata: &Metadata,
 ) -> Result<Transaction<T>, Error> {
     // 1. Validate this call against the current node metadata if the call comes
     // with a hash allowing us to do so.
     validate(call, metadata)?;
 
-    // 2. Encode extrinsic
+    // 3. Encode extrinsic
```

Comment:
```suggestion
    // 2. Encode extrinsic
```

revert, looks like it was correct before